### PR TITLE
Feature logan demo 18

### PR DESCRIPTION
**Pull request by logan858**
### What this PR does (required):
- Creates a demo user login button on login and signup pages

### Screenshots / Videos (required):
image
image
image

### Any information needed to test this feature (required):
- install dotenv in the client directory.  npm i || npm i dotenv@8.2.0
- create your .env file in client directory
- start the webapp and signup a new account.  You can use any to test, however, for demo purposes I used:  
> demoUser@email.com // deus123
> username: Demo
- add these lines (or whatever info you used for your demo account) to your .env file:
> REACT_APP_DEMO_USER=demoUser@email.com
> REACT_APP_DEMO_PW=deus123
- npm run build, this will allow access to env REACT_APP_ vars on the client side
- (optional) you can then add dummy data in the form of Requests documents to your database, however, ticket #31 is not merged into main yet, so your bookings page will not yet be able to reflect that db data.  we can further build out the demo user account with out-going requests once that is fully implemented and possibly a custom demo avatar, profile info, etc.

### CODE REVIEW

File: client/.eslintrc [hunk 1]
```diff
@@ -18,6 +18,8 @@
    "react/react-in-jsx-scope": "off",
    "react/prop-types": "off",
    "react-hooks/rules-of-hooks": "error",
-   "react-hooks/exhaustive-deps": "error"
+   "react-hooks/exhaustive-deps": "error",
+   // line formatting => comment out if causing problems
```
Comment (ezeilo-su) [inline client/.eslintrc:22]: Remove comments

File: client/src/pages/Bookings/Bookings.tsx [hunk 2]
```diff
@@ -14,7 +14,8 @@ import { ReactElement, useState } from 'react';
 export default function Bookings(): JSX.Element {
   const classes = useStyles();
   const [selectedDate, handleDateChange] = useState<Date>(new Date());
-  const dateTester = useState<Date>(new Date('09/13/2021'));
+  //TODO : with implementing on ticket 31
+  // const dateTester = useState<Date>(new Date('09/13/2021'));
```
Comment (ezeilo-su) [inline client/src/pages/Bookings/Bookings.tsx:18]: Move to your todo list

File: client/src/pages/SignUp/SignUp.tsx [hunk 3]
```diff
@@ -29,6 +31,26 @@ export default function Register(): JSX.Element {
         updateLoginContext(data.success);
       } else {
         // should not get here from backend but this catch is for an unknown issue
+
+        console.error({ data });
```
Comment (ezeilo-su) [inline client/src/pages/SignUp/SignUp.tsx:35]: No console message

File: client/src/pages/SignUp/SignUp.tsx [hunk 4]
```diff
@@ -29,6 +31,26 @@ export default function Register(): JSX.Element {
...
+    { setSubmitting }: FormikHelpers<{ email: string; password: string }>,
+  ) => {
+    login(email, password).then((data) => {
+      if (data.error) {
+        setSubmitting(false);
+        updateSnackBarMessage(data.error.message);
+      } else if (data.success) {
+        updateLoginContext(data.success);
+      } else {
+        // should not get here from backend but this catch is for an unknown issue
```
Comment (ezeilo-su) [inline client/src/pages/SignUp/SignUp.tsx:53]: Remove comments and console messages

**Review by ezeilo-su — COMMENTED** [review]
Nice work replicating the spec. Your UI is beautifully implemented. I just left a few comments for you to address.
Happy coding!

File: client/.eslintrc (hunk 1 above)
Comment (logan858) [reply client/.eslintrc]: removed comments

File: client/src/pages/SignUp/SignUp.tsx (hunk 3 above)
Comment (logan858) [reply client/src/pages/SignUp/SignUp.tsx]: removed comments & message

File: client/src/pages/SignUp/SignUp.tsx (hunk 4 above)
Comment (logan858) [reply client/src/pages/SignUp/SignUp.tsx]: removed comments & message

File: client/src/pages/SignUp/SignUp.tsx [hunk 5]
```diff
@@ -28,9 +30,23 @@ export default function Register(): JSX.Element {
...
       } else {
-        // should not get here from backend but this catch is for an unknown issue
-        console.error({ data });
+        setSubmitting(false);
+        updateSnackBarMessage('An unexpected error occurred. Please try again');
+      }
+    });
+  };
 
+  const handleDemoLogin = (
```
Comment (rajivtitus) [inline client/src/pages/SignUp/SignUp.tsx:39]: Why don't we define this function within the demo login component? If we are rendering the Demo Login component in multiple places we would need to pass this down as a prop each time

File: client/src/pages/Login/LoginForm/LoginForm.tsx [hunk 6]
```diff
@@ -30,68 +30,90 @@ export default function Login({ handleSubmit }: Props): JSX.Element {
...
+            />
+            <Box textAlign="center">
+              <Button type="submit" size="large" variant="contained" color="secondary" className={classes.submit}>
+                {isSubmitting ? <CircularProgress style={{ color: 'white' }} /> : 'Login'}
+              </Button>
+            </Box>
+          </form>
+        )}
+      </Formik>
+      <Formik
```
Comment (rajivtitus) [inline client/src/pages/Login/LoginForm/LoginForm.tsx:100]: Why don't we render the demo login button here as well? :)

File: client/src/pages/SignUp/SignUp.tsx (hunk 5 above)
Comment (logan858) [reply client/src/pages/SignUp/SignUp.tsx]: moved the function to the DemoLogin component, and also moved the DemoLogin component out of the signup forms directory into it's own component folder.  Also updated the useStyles files attached to the demologin component in the login & signup forms.

Now importing function and component itself from the components directory.

File: client/src/pages/Login/LoginForm/LoginForm.tsx (hunk 6 above)
Comment (logan858) [reply client/src/pages/Login/LoginForm/LoginForm.tsx]: updated where the demo button is being rendered across both the login & signups to be more consistent.

File: client/src/pages/SignUp/SignUpForm/useStyles.ts [hunk 7]
```diff
@@ -6,29 +6,32 @@ const useStyles = makeStyles((theme) => ({
     marginTop: theme.spacing(1),
   },
   label: {
-    fontSize: 19,
-    color: 'rgb(0,0,0,0.4)',
+    fontSize: 16,
     paddingLeft: '5px',
+    textTransform: 'uppercase',
+    color: 'black',
+    fontFamily: 'Poppins-Bold',
```
Comment (rajivtitus) [inline client/src/pages/SignUp/SignUpForm/useStyles.ts:13]: Please avoid setting the fontFamily outside of MUI's theme, please see article below on setting the font within the theme:

Ref: https://blog.logrocket.com/3-ways-to-add-custom-fonts-to-your-material-ui-project/

**Review by rajivtitus — APPROVED** [review]
LGTM! Good work componentizing the demo login 👍🏽 

Please address pending comment and resolve conflicts before merging. Thanks!

File: client/src/pages/SignUp/SignUpForm/useStyles.ts (hunk 7 above)
Comment (logan858) [reply client/src/pages/SignUp/SignUpForm/useStyles.ts:13]: Absolutely.  I updated those changes in ticket #26 in relation to the signupform & loginform.  Will give those changes priority when merging this ticket.

**Pull request merged**